lib/measured/unit.rb

```diff
@@ -56,8 +56,8 @@ def inverse_conversion_amount
 
   private
 
-  def case_insensitive(comparison)
-    [comparison].flatten.map(&:downcase)
+  def downcased_names
+    @downcased_names ||= @names.map(&:downcase)
```

Comment:
Yep, you got it. It'll generally be pretty small, and if one arranges it to put the most common aliases first, it'll be even faster. I'm going to benchmark something else quickly.